.github/workflows/ci.yml

```diff
@@ -15,12 +15,20 @@ jobs:
             env: py36
           - distro: centos-8
             env: py38
+          - distro: centos-8
+            env: container
```

Comment:
So we double the number of jobs, one job will run the tests, and a second job will run the container tests?

Which version of the container is tested? If we test the container (e.g. alpine) why do we need to run the same tests 4 times with all distros? If the code under test is what we have in the container, we need to test it only once.

Maybe we need another job, running on ubuntu (without docker), building the imageio container (using docker or podman on ubuntu), starting the container using github yaml, and running the relevant tests (on ubuntu), accessing the running container?

But we have to think also about local testing - we can actually start with having only local testing - how a developer will run the tests locally?

I think we need to think more about the big picture and agree on how this should work:
- local testing
- ci testing
- real usage with podman/docker
- real usage in k8s

Reply:
```
Maybe we need another job, running on ubuntu (without docker), building the imageio container (using docker or podman on ubuntu), starting the container using github yaml, and running the relevant tests (on ubuntu), accessing the running container?
```
I considered that too, for the same reason you expose. So far, I am mostly testing what is needed to make it work (like checking dependencies, seeing if the podman approach works in the CIs, etc.). After that, I will refine this and probably follow the approach you mentioned.

Regarding the local test, it should be already possible to run local tests. I have run them. But I need to document it. And sure, let's have a discussion about it.

Reply:
OK, hit my head against a wall for a while with the ci testing…

Couldn't find a way to make it work so far. The test in this branch works locally, but not in the pipelines. So maybe we can start, as you mentioned, with only local testing. And after that I'll keep trying to make the ci testing run. We would need to discuss how do we want to achieve it.